test/integration/targets/inventory_aws_conformance/runme.sh

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# set the output dir
+if [ -z ${OUTPUT_DIR+null} ]; then
+    export OUTPUT_DIR=$(pwd)
+fi
+
+#################################################
+#   RUN THE SCRIPT
+#################################################
+
+# run the script first
+cat << EOF > $OUTPUT_DIR/ec2.ini
+[ec2]
+regions = us-east-1
+cache_path = $(pwd)/.cache
+cache_max_age = 0
+group_by_tag_none = False
+
+[credentials]
+aws_access_key_id = FOO
+aws_secret_acccess_key = BAR
+EOF
+
+rm -f script.out
+./ec2.sh | tee -a $OUTPUT_DIR/script.out
+#./ec2.sh 
+RC=$?
+if [[ $RC != 0 ]]; then
+    exit $RC
+fi
+#rm -f $OUTPUT_DIR/ec2.ini
+#rm -f $OUTPUT_DIR/ec2.py
+rm -rf .cache
+
+#exit 0
+
+#################################################
+#   RUN THE PLUGIN
+#################################################
+
+# run the plugin second
+export ANSIBLE_INVENTORY_ENABLED=aws_ec2
+export ANSIBLE_INVENTORY=test.aws_ec2.yml
+export AWS_ACCESS_KEY_ID=FOO
+export AWS_SECRET_ACCESS_KEY=BAR
+export ANSIBLE_TRANSFORM_INVALID_GROUP_CHARS=never
+
+cat << EOF > $OUTPUT_DIR/test.aws_ec2.yml
+plugin: aws_ec2
+cache: False
```

Comment:
The documented cache default is "no", so I intend to not provide the key when building syntax like this, let me know if I should change that practice.

I commented elsewhere about how I think it would be an easier task for you to just assert on the modern variables.

If you _are_ going to use this, then I think you should add `strict: True`. Otherwise you don't really know if this is or isn't working, and there will probably be surprises later on.

Reply:
Good point about `strict`, will fix that in my next commit.

Yes, cache will stay off by default.

I think there's some security in specificity of testing and benefit to testing the output of the script since core has no tests for it. And it may be helpful to users if there is more than an implicit test for how to replicate script behavior. We would like to test for all the script hostvars. Since I'm just adding onto @jctanner's inspiration I'll leave that decision up to him though since it would be taking this PR in a new direction.